### PR TITLE
 added account balance

### PR DESCRIPTION
**Pull request by manuelgeek**
## What is the Purpose?

The Account Balance API requests for the account balance of a short code.

### Configs
```elixir
    config :ex_pesa,
        mpesa: [
            cert: "",
            balance: [
                initiator_name: "Safaricom1",
                password: "Safaricom133",
                timeout_url: "",
                result_url: "",
                security_credential: ""
            ]
        ]
  ```
then ... 
`ExPesa.Mpesa.AccountBalance.request(%{command_id: "AccountBalance", short_code: "602843", remarks: "remarks", account_type: "Customer"})`

### CODE REVIEW

File: lib/ex_pesa/Mpesa/account_balance.ex [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        account_type: account_type
+      }) do
+    payload = %{
+      "Initiator" => Application.get_env(:ex_pesa, :mpesa)[:balance][:initiator_name],
+      "SecurityCredential" => security_credential,
+      "CommandID" => command_id,
+      "PartyA" => short_code,
+      "IdentifierType" => 4,
+      "Remarks" => remarks,
+      "AccountType" => account_type,
```
Comment (lenileiro) [inline lib/ex_pesa/Mpesa/account_balance.ex:87]: I don't think the account_type field is required when checking account balance. I think that might be a typo on the daraja docs.

File: lib/ex_pesa/Mpesa/account_balance.ex [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    end
+  end
+
+  @doc false
+  def request() do
+    {:error,
+     "Required Parameter missing, 'command_id','short_code', 'remarks','account_reference'"}
+  end
+
+  def account_balance(security_credential, %{
```
Comment (lenileiro) [inline lib/ex_pesa/Mpesa/account_balance.ex:74]:
```suggestion
  defp account_balance(security_credential, %{
```

File: lib/ex_pesa/Mpesa/account_balance.ex (hunk 1 above)
Comment (lenileiro) [inline lib/ex_pesa/Mpesa/account_balance.ex:87]:
```suggestion
```

File: lib/ex_pesa/Mpesa/account_balance.ex [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        command_id: command_id,
+        short_code: short_code,
+        remarks: remarks,
+        account_type: account_type
```
Comment (lenileiro) [inline lib/ex_pesa/Mpesa/account_balance.ex:78]:
```suggestion
       
```

File: lib/ex_pesa/Mpesa/account_balance.ex [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  attrs: - a map containing:
+  - `command_id` - A unique command passed to the M-Pesa system..
+  - `short_code` - The shortcode of the organisation receiving the transaction.
+  - `remarks` - Comments that are sent along with the transaction.
+  - `account_type` - Organisation receiving the funds.
+
+  ## Example
+
+      iex> ExPesa.Mpesa.AccountBalance.request(%{command_id: "AccountBalance", short_code: "602843", remarks: "remarks", account_type: "Customer"})
```
Comment (lenileiro) [inline lib/ex_pesa/Mpesa/account_balance.ex:51]:
```suggestion
      iex> ExPesa.Mpesa.AccountBalance.request(%{command_id: "AccountBalance", short_code: "602843", remarks: "Checking account balance"})
```

File: lib/ex_pesa/Mpesa/account_balance.ex [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  def account_balance(security_credential, %{
+        command_id: command_id,
+        short_code: short_code,
+        remarks: remarks,
+        account_type: account_type
+      }) do
+    payload = %{
+      "Initiator" => Application.get_env(:ex_pesa, :mpesa)[:balance][:initiator_name],
+      "SecurityCredential" => security_credential,
+      "CommandID" => command_id,
```
Comment (lenileiro) [inline lib/ex_pesa/Mpesa/account_balance.ex:83]:
```suggestion
      "CommandID" => "AccountBalance",
```
@manuelgeek I think you can just default the command_id to AccountBalance

Comment (manuelgeek) [reply lib/ex_pesa/Mpesa/account_balance.ex]: awesome

**Comment by manuelgeek** [comment]
new test code `ExPesa.Mpesa.AccountBalance.request()` 
awesome work @lenileiro

**Review by lenileiro — APPROVED** [review]
LGTM

**Comment by manuelgeek** [comment]
@kamalogudah  kindly review

**Review by kamalogudah — APPROVED** [review]
LGTM

**Pull request merged**